@Test(expected = CommitFailedException.class)
public void testUniqueConstraint() throws Exception {
    SyncResult res = syncCtx.sync(idp.getUser(USER_ID));
    try {
        Tree t = r.getTree(getTestUser().getPath());
        t.setProperty(DefaultSyncContext.REP_EXTERNAL_ID, res.getIdentity().getExternalIdRef().getString());
        r.commit();
        fail("Duplicate value for rep:externalId must be detected in the default setup.");
    } catch (CommitFailedException e) {
        // success: verify nature of the exception
        assertException(e, CONSTRAINT, 30);
    } finally {
        r.refresh();
    }
}
